include/SSkinMeshBuffer.h

```diff
@@ -328,7 +328,22 @@ struct SSkinMeshBuffer : public IMeshBuffer
 	}
 
 	//! append the vertices and indices to the current buffer
-	void append(const void* const vertices, u32 numVertices, const u16* const indices, u32 numIndices) override {}
+	void append(const void* const vertices, u32 numVertices, const u16* const indices, u32 numIndices) override {
+		if (vertices == getVertices())
+			return;
+
+		const u32 vertexCount = getVertexCount();
+		Vertices_Standard.reallocate(vertexCount + numVertices);
```

Comment:
`getIndexCount()` could be inlined like it was for `getIndexCount()` on line 242.

Reply:
See https://github.com/JosiahWI/irrlicht/pull/14#discussion_r1459504280, I'll probably get rid of this entirely.

Reply:
You mean `getVertexCount`? The `vertexCount` is actually used twice (but I still wouldn't be necessarily opposed to inlining it).

Reply:
The name of the temporary doesn't do much for readability since the getter is already simple. Unless `getVertexCount()` is an expensive call, which seems unlikely, it's probably fine to inline it in both places.

Reply:
It isn't really expensive (just a switch, but one call is in a loop, so I'd still be slightly wary, though I assume any half decent compile could convince itself that the value is effectively constant in the loop), *but it is mutated through side effects* after we have added the vertices to `Vertices_Standard`. So we can't inline it.

We could swap the two code blocks around (first set the indices) and then we could inline it, but that would seem error prone as far as future maintainability is concerned. The variable draws attention to the issue; I've also added a comment (oh, and a buffer type check: for non-vertex mesh buffers, we want to throw an error).

Reply:
Good clarification by renaming the variable.